auth/forms.py

```diff
@@ -4,13 +4,25 @@
 
 from backend.models import UserCourse, Course
 from django.contrib.auth.models import Group
+from django.conf import settings
 
 
 class SignupForm(forms.Form):
 
+    def clean(self):
+        super(SignupForm, self).clean()
+        course_codes = self.data.get('course_code').replace(' ', '').split(';')
+        for course_code in course_codes:
+            #if the course code is empty or is neither an instructor nor a valid student course code
+            if not course_code or (course_code != settings.INSTRUCTOR_CODE and
+                                           Course.objects.filter(code=course_code).count() < 1):
```

Comment:
I think it probably makes more sense to split the instructor logic above the loop.  i.e. before anything just have:
```python
if len(course_codes) == 1 and course_code == settings.INSTRUCTOR_CODE:
    return
```
or do we want someone to be able to have multiple course codes with an instructor code mixed in?